test/Microsoft.TemplateEngine.IDE.IntegrationTests/End2EndTests.cs

```diff
@@ -423,6 +424,33 @@ internal async Task Test_CreateAsync_PreferDefaultNameInvalidParameters()
                 result.ErrorMessage);
         }
 
+        [Fact]
+        internal async Task PostAction_WithFileRename_Test()
+        {
+            using Bootstrapper bootstrapper = GetBootstrapper();
+            string templateLocation = GetTestTemplateLocation("PostActions/WithFileRename");
+            await InstallTemplateAsync(bootstrapper, templateLocation).ConfigureAwait(false);
+
+            string output = TestUtils.CreateTemporaryFolder();
+            IReadOnlyList<ITemplateMatchInfo> foundTemplates = await bootstrapper
+                .GetTemplatesAsync(new[] { WellKnownSearchFilters.NameFilter("TestAssets.PostActions.AddJsonProperty.WithSourceNameChangeInJson") })
+                .ConfigureAwait(false);
+
+            // Using this parameter with no real info so bootstrapper.CreateAsync is not an ambiguous call
+            Dictionary<string, string?> parameters = new();
+
+            ITemplateCreationResult result = await bootstrapper
+                .CreateAsync(foundTemplates[0].Info, "CompanyProject", output, parameters)
+                .ConfigureAwait(false);
+
+            IPostAction postAction = Assert.Single(result.CreationResult!.PostActions);
+            Assert.Equal("testfile.json", postAction.Args["jsonFileName"]);
+            Assert.Equal("moduleConfiguration:edgeAgent:properties.desired:modules", postAction.Args["parentPropertyPath"]);
+            Assert.Equal("CompanyProject", postAction.Args["newJsonPropertyName"]);
+            Assert.Equal("${MODULEDIR<../CompanyProject>}", postAction.Args["newJsonPropertyValue"]);
+            Assert.Equal("Add CompanyProject property to testfile.json manually.", postAction.ManualInstructions);
```

Comment:
When you use t he modified args in the PostAction (in this case the `newJsonPropertyName` for instance, will this change be available in the custom action then ? ).

I mean, I suppose that the renames to those args occur before the code of the custom PostAction is executed ?